compiler/qsc_partial_eval/src/lib.rs

```diff
@@ -1912,6 +2081,223 @@ impl<'a> PartialEvaluator<'a> {
+            (bin_op_rir_variable, bin_op_rir_ins)
+        }
+        BinOp::Shl => {
+            let bin_op_rir_variable = rir::Variable::new_integer(bin_op_variable_id);
+            let bin_op_rir_ins = Instruction::Shl(lhs_operand, rhs_operand, bin_op_rir_variable);
+            (bin_op_rir_variable, bin_op_rir_ins)
+        }
+        BinOp::Shr => {
+            let bin_op_rir_variable = rir::Variable::new_integer(bin_op_variable_id);
+            let bin_op_rir_ins = Instruction::Ashr(lhs_operand, rhs_operand, bin_op_rir_variable);
+            (bin_op_rir_variable, bin_op_rir_ins)
+        }
+        BinOp::Eq => {
+            let bin_op_rir_variable = rir::Variable::new_boolean(bin_op_variable_id);
+            let bin_op_rir_ins = Instruction::Icmp(
+                ConditionCode::Eq,
+                lhs_operand,
+                rhs_operand,
+                bin_op_rir_variable,
+            );
+            (bin_op_rir_variable, bin_op_rir_ins)
+        }
+        BinOp::Neq => {
+            let bin_op_rir_variable = rir::Variable::new_boolean(bin_op_variable_id);
+            let bin_op_rir_ins = Instruction::Icmp(
+                ConditionCode::Ne,
+                lhs_operand,
+                rhs_operand,
+                bin_op_rir_variable,
+            );
+            (bin_op_rir_variable, bin_op_rir_ins)
+        }
+        BinOp::Gt => {
+            let bin_op_rir_variable = rir::Variable::new_boolean(bin_op_variable_id);
+            let bin_op_rir_ins = Instruction::Icmp(
+                ConditionCode::Sgt,
+                lhs_operand,
+                rhs_operand,
+                bin_op_rir_variable,
+            );
+            (bin_op_rir_variable, bin_op_rir_ins)
+        }
+        BinOp::Gte => {
+            let bin_op_rir_variable = rir::Variable::new_boolean(bin_op_variable_id);
+            let bin_op_rir_ins = Instruction::Icmp(
+                ConditionCode::Sge,
+                lhs_operand,
+                rhs_operand,
+                bin_op_rir_variable,
+            );
+            (bin_op_rir_variable, bin_op_rir_ins)
+        }
+        BinOp::Lt => {
+            let bin_op_rir_variable = rir::Variable::new_boolean(bin_op_variable_id);
+            let bin_op_rir_ins = Instruction::Icmp(
+                ConditionCode::Slt,
+                lhs_operand,
+                rhs_operand,
+                bin_op_rir_variable,
+            );
+            (bin_op_rir_variable, bin_op_rir_ins)
+        }
+        BinOp::Lte => {
+            let bin_op_rir_variable = rir::Variable::new_boolean(bin_op_variable_id);
+            let bin_op_rir_ins = Instruction::Icmp(
+                ConditionCode::Sle,
+                lhs_operand,
+                rhs_operand,
+                bin_op_rir_variable,
+            );
+            (bin_op_rir_variable, bin_op_rir_ins)
+        }
+        _ => panic!("unsupported binary operation for integers: {bin_op:?}"),
+    };
+    Ok((rir_variable, rir_ins))
```

Comment:
nitpick: this could be a wrapping `Ok()` around the match expression, and then you don't need to let-bind the variables. But that may just be a personal taste thing, so feel free to ignore.

Reply:
I usually prefer it the other way 😅. `match` expressions, especially long ones, binding to a variable before wrapping it in an `Ok()`. I'll leave this one as is 😄